### PR TITLE
new layout

### PR DESCRIPTION
**Pull request by tom1903**
changed the old <textview> and <image view> in the phone ui to checkboxes.

also added a sports display to the CarDetailedInfo

### CODE REVIEW

File: app/src/main/res/layout/fragment_overview_connected.xml [hunk 1]
```diff
@@ -19,6 +19,7 @@
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
             android:id="@+id/fragmentCarSummary"
+            android:background="@color/red"
```
Comment (hufman) [inline app/src/main/res/layout/fragment_overview_connected.xml:22]: Leftover debugging, please remove

File: .idea/misc.xml [hunk 2]
```diff
@@ -1,6 +1,5 @@
-<?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_11" default="true" project-jdk-name="JDK" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_15" default="true" project-jdk-name="15" project-jdk-type="JavaSDK">
```
Comment (hufman) [inline .idea/misc.xml:2]: What does this change do? I'm leery of upgrading the JDK in case it breaks compatibility for the old OS versions the app supports. I was able to compile and run it without this change.

File: .idea/kotlinc.xml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="KotlinJpsPluginSettings">
```
Comment (hufman) [inline .idea/kotlinc.xml:3]: What is this change?

File: .idea/deploymentTargetDropDown.xml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (hufman) [inline .idea/deploymentTargetDropDown.xml:1]: This file references your phone and should not be included.

File: app/src/main/java/me/hufman/androidautoidrive/phoneui/viewmodels/DependencyInfoModel.kt [hunk 5]
```diff
@@ -78,6 +78,8 @@ class DependencyInfoModel(val connection: CarConnectionDebugging, val interrupti
 		_isSecurityServiceDisconnected.value = connection.isConnectedSecurityInstalled && !connection.isConnectedSecurityConnected
 		_hasBackgroundKilled.value = interruptionDetection.detectedKilled >= 3
 		_hasBackgroundSuspended.value = interruptionDetection.detectedSuspended >= 2
+
```
Comment (hufman) [inline app/src/main/java/me/hufman/androidautoidrive/phoneui/viewmodels/DependencyInfoModel.kt:81]: Unnecessary whitespace change

**Review by hufman — CHANGES_REQUESTED** [review]
Thanks for the PR!
Can you tell me the story of the motivation for this PR? I like how it simplifies the layout files, but this seems like a lot of work for you to work on for little benefit :)

I like the little animation when the checkboxes come into view. However, the unchecked checkboxes look too clickable, can you also toggle their enabled state to make them look inactive while unchecked?

Can you split it into two PRs (Sports Displays and then PhoneUI changes), so that the two separate features can be merged as two separate commits?

You unfortunately committed your Gradle properties (even in the .idea unstashed changes file), so your keys are now public and will need to be revoked.

**Comment by tom1903** [comment]
Hi! My idea was just that the layout looks a little different and I like checkboxes ;) But if it doesn't fit into your original concept of the app, then decline the PR.
I just like programming, it relaxes me after a long day, so all good! :)

I guess that it would be possible to make the checkboxes look inactive while unchecked. I will check this.

Tomorrow I will separate the two features.

That I passed the grandle properties is a little bit unfortunate... I totally overlooked ... Thank you!

File: .idea/misc.xml (hunk 2 above)
Comment (tom1903) [reply .idea/misc.xml:2]: That's just a change for my IDE. For.some reason I can only compile when it's set to JDK 15
In the next pr I will review in more detail so the unnecessary files won't be pushed

**Comment by hufman** [comment]
I'm fine with the idea, and I look forward to the next PRs :)
The Sports Displays overlap a little with the Driving Information screen, feel free to move some of the more technical fields over to the Sports Displays. There's also a [CDS value with the torque output](https://bimmergestalt.github.io/BMWConnectedAnalysis/cds/) which would be fun to add too!

File: app/src/main/java/me/hufman/androidautoidrive/carapp/carinfo/CarDetailedInfo.kt [hunk 6]
```diff
@@ -164,6 +166,13 @@ class CarDetailedInfo(carCapabilities: Map<String, Any?>, cdsMetrics: CDSMetrics
 	val rawHeading = cdsMetrics.rawHeading.format("%.0f Raw Heading")
 
 	// categories
+	private val sportFields: List<Flow<String>> = listOf(
+			engineTemp, oilTemp,
+			accelContact, brakeState,
+			clutchState, steeringAngle,
+			tempInterior, tempExterior,
+			gforces, emptyFlow()
```
[the remark this answers is not in the document]
Comment (tom1903) [reply app/src/main/java/me/hufman/androidautoidrive/carapp/carinfo/CarDetailedInfo.kt:174]: Do you know if it is possible to show the geforces in the whole line? So if I use the splitscreen in my bmw, only the first half of the geforces is visible.

**Comment by tom1903** [comment]
I will close this PR and make new ones for each feature

File: app/src/main/java/me/hufman/androidautoidrive/carapp/carinfo/CarDetailedInfo.kt (hunk 6 above)
[the remark this answers is not in the document]
Comment (hufman) [reply app/src/main/java/me/hufman/androidautoidrive/carapp/carinfo/CarDetailedInfo.kt:174]: Weird, in my Mini 8" screen, it shows fine in splitscreen mode.
But sure, you could split up the gforces:
```
val gforces = cdsMetrics.accel.map { accel ->
	val lat = accel.first?.let {"↔%.2f".format(it/9.8)} ?: ""
	val long = accel.second?.let {"↕%.2f".format(it/9.8)} ?: ""
	"$lat $long ${L.CARINFO_GFORCE}"
}
val gforceLat = cdsMetrics.accel.map { accel ->
	val lat = accel.first?.let {"↔%.2f".format(it/9.8)} ?: ""
	"$lat"
}
val gforceLong = cdsMetrics.accel.map { accel ->
	val long = accel.second?.let {"↕%.2f".format(it/9.8)} ?: ""
	"$long ${L.CARINFO_GFORCE}"
}
```
